File: 卡码/代码随想录/1/09-动态规划/Python/05-子序列问题/03-编辑距离/LeetCode_0392.py
#392. 判断子序列
from typing import *

class Solution:
  def isSubsequence(self, s: str, t: str) -> bool:
    dp:List[List[int]]=[[0]*(len(t)+1) for _ in range(len(s)+1)]
    for i in range(1,len(s)+1):
      for j in range(1,len(t)+1):
        if s[i-1]==t[j-1]:
          dp[i][j]=dp[i-1][j-1]+1
        else:
          dp[i][j]=dp[i][j-1]
    return dp[len(s)][len(t)]==len(s)

File: 卡码/代码随想录/1/09-动态规划/Js/05-子序列问题/03-编辑距离/LeetCode_0392.js
//392. 判断子序列
const isSubsequence = function(s, t) {
	const dp=new Array(s.length+1).fill(0).map(item=>new Array(t.length+1).fill(0));
	for(let i=1;i<s.length+1;i++){
		for(let j=1;j<t.length+1;j++){
			if(s[i-1]==t[j-1]) dp[i][j]=dp[i-1][j-1]+1;
			else dp[i][j]=dp[i][j-1];
		}
	}
	return dp[s.length][t.length]==s.length;
};
